lib/integration/framework/Test/Integration/Framework/Setup.hs

```diff
@@ -504,8 +505,35 @@ withContext testingCtx@TestingCtx{..} action = do
                     nodeOutputFile
                     cluster
                 )
-                (takeMVar ctx >>= bracketTracer' tr "spec" . action)
+                (takeMVar ctx >>= bracketTracer' tr "spec" .
+                    (\c -> setupDelegation faucetClientEnv c >> action c))
         whenLeft res (throwIO . ProcessHasExited "integration")
+  where
+    -- | Setup delegation for 'rewardWallet' / 'rewardWalletMnemonics'.
+    --
+    -- Rewards take 4-5 epochs (here ~2 min) to acrue from when we delegate. By
+    -- doing this up-front, the rewards are likely available by the time
+    -- 'rewardWallet' is called, and we save time.
+    setupDelegation :: ClientEnv -> Context -> IO ()
+    setupDelegation faucetClientEnv ctx = do
+        mnemonics <- runFaucetM faucetClientEnv Faucet.rewardWalletMnemonics
+        pool : _ : _ <-
```

Comment:
```suggestion
        pool <-
```
+ head probably

Reply:
doesn't that ie. `pool:_:_` enforce that there are 3 pools we are expecting? If yes, then I would argue `pool:_:_`is fine